File: backend/app/schemas/storage.py
"""
스토리지 관리 스키마
"""
from typing import Optional, List, Dict
from pydantic import BaseModel


class StorageCategoryStats(BaseModel):
    """스토리지 카테고리별 통계"""
    size_bytes: int
    count: int


class StorageStatsResponse(BaseModel):
    """스토리지 통계 응답"""
    bucket_name: str
    total_size_bytes: int
    total_objects: int
    categories: Dict[str, StorageCategoryStats]
    error: Optional[str] = None


class OrphanedFile(BaseModel):
    """고아 파일 정보"""
    object_name: str
    size: int
    last_modified: Optional[str] = None


class OrphanedFilesResponse(BaseModel):
    """고아 파일 목록 응답"""
    count: int
    total_size_bytes: int
    files: List[OrphanedFile]


class CleanupError(BaseModel):
    """정리 오류"""
    object_name: Optional[str] = None
    error: str


class CleanupResponse(BaseModel):
    """정리 결과 응답"""
    deleted_count: int
    deleted_size_bytes: int
    errors: List[CleanupError]

File: frontend/src/types/storage.ts
export interface StorageCategoryStats {
  size_bytes: number;
  count: number;
}

export interface StorageStatsResponse {
  bucket_name: string;
  total_size_bytes: number;
  total_objects: number;
  categories: {
    documents: StorageCategoryStats;
    pages: StorageCategoryStats;
    thumbnails: StorageCategoryStats;
    other: StorageCategoryStats;
  };
  error?: string;
}

export interface OrphanedFile {
  object_name: string;
  size: number;
  last_modified?: string;
}

export interface OrphanedFilesResponse {
  count: number;
  total_size_bytes: number;
  files: OrphanedFile[];
}

export interface CleanupResponse {
  deleted_count: number;
  deleted_size_bytes: number;
  errors: Array<{
    object_name?: string;
    error: string;
  }>;
}
